code/modules/food/food/condiment.dm

```diff
@@ -441,30 +453,87 @@
 
 //End of MRE stuff.
 
-/obj/item/weapon/reagent_containers/food/condiment/flour
-	name = "flour sack"
-	desc = "A big bag of flour. Good for baking!"
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour
+	name = "flour carton"
+	desc = "A big carton of flour. Good for baking!"
 	icon = 'icons/obj/food.dmi'
 	icon_state = "flour"
 	volume = 220
 	center_of_mass = list("x"=16, "y"=8)
 
-/obj/item/weapon/reagent_containers/food/condiment/flour/on_reagent_change()
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/on_reagent_change()
+	update_icon()
 	return
 
-/obj/item/weapon/reagent_containers/food/condiment/flour/Initialize()
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/Initialize()
 	. = ..()
 	reagents.add_reagent("flour", 200)
 	randpixel_xy()
 
+/obj/item/weapon/reagent_containers/food/condiment/advanced/update_icon()
+	overlays.Cut()
+
+	if(reagents.total_volume)
+		var/image/filling = image('icons/obj/food.dmi', src, "[icon_state]10")
+
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(-1 to 9)		filling.icon_state = "[icon_state]-10"
+			if(10 to 24) 	filling.icon_state = "[icon_state]-25"
+			if(25 to 49)	filling.icon_state = "[icon_state]-50"
+			if(50 to 85)	filling.icon_state = "[icon_state]-75"
+			if(86 to INFINITY)	filling.icon_state = "[icon_state]-100"
+
+		overlays += filling
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/rustic
+	name = "flour sack"
+	desc = "An artisanal sack of flour. Classy!"
+	icon_state = "flour_bag"
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/rustic/empty
+
+/obj/item/weapon/reagent_containers/food/condiment/advanced/flour/rustic/empty/Initialize()
+	. = ..()
+	reagents.remove_reagent("flour", 200)
```

Comment:
If you need to make subtypes for "empty", just make a `starts_with` list that specifies what reagents the container gets initialized with.

Reply:
This would, I imagine, cull a lot of copy-pasted initialize() functions in condiment defines.

Reply:
It'd be novel functionality, aye. Sorta like how lockers work, you make a generic `starts_with` list on reagent containers, and upon initialization, after creating the `reagents` holder datum, you iterate over the `starts_with` list and add each reagent specified therein in the proscribed amount. Something like

```
/obj/item/weapon/reagent_containers
	var/list/starts_with_reagents = list(
		"water" = 20,
		"wine" = 30
	)

/obj/item/weapon/reagent_containers/Initialize()
	. = ..() // reagents is created higher in the chain
	if(LAZYLEN(starts_with_reagents))
		for(var/reagent in starts_with_reagents)
			reagents.add_reagent(reagent, starts_with_reagents[reagent])
```